### ISSUE TITLE
Using closure as argument causes “Multi-line function definition not indented correctly”

### ISSUE THREAD

**Issue by westonruter** [squizlabs/PHP_CodeSniffer#318]
I've identified what I think is a problem in `PEAR.Functions.FunctionDeclaration`:

![image](https://cloud.githubusercontent.com/assets/134745/4991373/d6914da0-69e6-11e4-87d3-317b1dd5fe32.png)

This seems like it should be valid, and that the expected indent is not getting incremented by one unit when inside of a closure.

**Comment by GaryJones** [comment]
I think there may still be an issue here:

![screen shot 2015-07-15 at 17 23 11](https://cloud.githubusercontent.com/assets/88371/8703654/4e03e56e-2b16-11e5-82c3-fa7767fed296.png)

~~~php
			array_walk( $upcoming_webinars, function( $webinar ) {
				$webinar_date = DateTime::createFromFormat( 'Ymd', get_field( 'event_date', $webinar->ID ) );
				$webinar_time = get_field( 'event_time', $webinar->ID );
				?>
						<tr>
							<td><?php echo get_the_title( $webinar->ID ); ?></td>
							<td><?php echo $webinar_date->format( 'Y-m-d' ); ?></td>
							<td><?php echo $webinar_time; ?></td>
							<td><a href="<?php echo esc_url( get_permalink( $webinar->ID ) ); ?>"><strong><?php esc_html_e( 'Register Now', 'danielstrading' ); ?></strong></a></td>
						</tr>
				<?php
			});
~~~

Leads to:

~~~
 347 | ERROR | [x] Multi-line function call not indented correctly; expected 12 spaces but found 0
     |       |     (PEAR.Functions.FunctionCallSignature.Indent)
 348 | ERROR | [x] Multi-line function call not indented correctly; expected 12 spaces but found 0
     |       |     (PEAR.Functions.FunctionCallSignature.Indent)
 349 | ERROR | [x] Multi-line function call not indented correctly; expected 12 spaces but found 0
     |       |     (PEAR.Functions.FunctionCallSignature.Indent)
 350 | ERROR | [x] Multi-line function call not indented correctly; expected 12 spaces but found 0
     |       |     (PEAR.Functions.FunctionCallSignature.Indent)
 351 | ERROR | [x] Multi-line function call not indented correctly; expected 12 spaces but found 0
     |       |     (PEAR.Functions.FunctionCallSignature.Indent)
 352 | ERROR | [x] Multi-line function call not indented correctly; expected 12 spaces but found 0
     |       |     (PEAR.Functions.FunctionCallSignature.Indent)
 353 | ERROR | [x] Multi-line function call not indented correctly; expected 12 spaces but found 0
     |       |     (PEAR.Functions.FunctionCallSignature.Indent)
~~~

I can't seem to shift that table row to any indentation to stop the errors from appearing.

**Comment by GaryJones** [comment]
A simpler snippet, giving the same error:

~~~php
			array_walk( $types, function( $title, $type ) {
				$plural = 'all' !== $type ? 's' : '';
				?>
				<li><a href="<?php esc_url( '#tabs-documents-' . $type . $plural ); ?>"><?php echo esc_html( $title ); ?></a></li>
				<?php
			});
~~~

**Comment by gsherwood** [comment]
@GaryJones That's because the sniff doesn't support inline HTML like that. I've committed a fix here: https://github.com/squizlabs/PHP_CodeSniffer/commit/6a2629ab641ea9169dbc34276e19cb4f2dd3a287

**Comment by djereg** [comment]
Hi! I have the same multi-line error when passing closure as function argument, but without HTML content. 

```php
array_reduce($array, function ($carry, $item) use ($variable) {
    return true;
});
```

`phpcs -v -s app/Helpers/test.php`
```
Registering sniffs in the pcsg-generated-ruleset standard... DONE (60 sniffs registered)
Creating file list... DONE (1 files in queue)
Changing into directory G:\...\app\Helpers
Processing test.php [PHP => 34 tokens in 5 lines]... DONE in 1ms (3 errors, 0 warnings)

FILE: G:\...\app\Helpers\test.php
----------------------------------------------------------------------
FOUND 3 ERRORS AFFECTING 2 LINES
----------------------------------------------------------------------
 3 | ERROR | [x] Opening parenthesis of a multi-line function call
   |       |     must be the last content on the line
   |       |     (PEAR.Functions.FunctionCallSignature.ContentAfterOpenBracket)
 3 | ERROR | [x] Only one argument is allowed per line in a
   |       |     multi-line function call
   |       |     (PEAR.Functions.FunctionCallSignature.MultipleArguments)
 5 | ERROR | [x] Closing parenthesis of a multi-line function call
   |       |     must be on a line by itself
   |       |     (PEAR.Functions.FunctionCallSignature.CloseBracketLine)
----------------------------------------------------------------------
PHPCBF CAN FIX THE 3 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------

Time: 106ms; Memory: 8Mb
```

Or maybe I was missed something?

`phpcs -vv -s app/Helpers/test.php`
```
Processing ruleset G:\...\phpcs.xml
        Processing rule "Generic.Classes.DuplicateClassName"
                => G:\...\vendor\squizlabs\php_codesniffer\src\Standards\Generic\Sniffs\Classes\DuplicateClassNameSniff.php
        Processing rule "Generic.CodeAnalysis.EmptyStatement"
                => G:\...\vendor\squizlabs\php_codesniffer\src\Standards\Generic\Sniffs\CodeAnalysis\EmptyStatementSniff.php
        Processing rule "Generic.CodeAnalysis.ForLoopShouldBeWhileLoop"
                => G:\...\vendor\squizlabs\php_codesniffer\src\Standards\Generic\Sniffs\CodeAnalysis\ForLoopShouldBeWhileLoopSniff.php
        Processing rule "Generic.CodeAnalysis.ForLoopWithTestFunctionCall"
                => G:\...\vendor\squizlabs\php_codesniffer\src\Standards\Generic\Sniffs\CodeAnalysis\ForLoopWithTestFunctionCallSniff.php
        Processing rule "Generic.CodeAnalysis.JumbledIncrementer"
                => G:\...\vendor\squizlabs\php_codesniffer\src\Standards\Generic\Sniffs\CodeAnalysis\JumbledIncrementerSniff.php
        Processing rule "Generic.CodeAnalysis.UnconditionalIfStatement"
                => G:\...\vendor\squizlabs\php_codesniffer\src\Standards\Generic\Sniffs\CodeAnalysis\UnconditionalIfStatementSniff.php
        Processing rule "Generic.CodeAnalysis.UnnecessaryFinalModifier"
                => G:\...\vendor\squizlabs\php_codesniffer\src\Standards\Generic\Sniffs\CodeAnalysis\UnnecessaryFinalModifierSniff.php
        Processing rule "Generic.ControlStructures.InlineControlStructure"
                => G:\...\vendor\squizlabs\php_codesniffer\src\Standards\Generic\Sniffs\ControlStructures\InlineControlStructureSniff.php
        Processing rule "Generic.Files.ByteOrderMark"
                => G:\...\vendor\squizlabs\php_codesniffer\src\Standards\Generic\Sniffs\Files\ByteOrderMarkSniff.php
        Processing rule "Generic.Formatting.DisallowMultipleStatements"
                => G:\...\vendor\squizlabs\php_codesniffer\src\Standards\Generic\Sniffs\Formatting\DisallowMultipleStatementsSniff.php
        Processing rule "Generic.Formatting.NoSpaceAfterCast"
                => G:\...\vendor\squizlabs\php_codesniffer\src\Standards\Generic\Sniffs\Formatting\NoSpaceAfterCastSniff.php
        Processing rule "Generic.Functions.CallTimePassByReference"
                => G:\...\vendor\squizlabs\php_codesniffer\src\Standards\Generic\Sniffs\Functions\CallTimePassByReferenceSniff.php
        Processing rule "Generic.Functions.OpeningFunctionBraceBsdAllman"
                => G:\...\vendor\squizlabs\php_codesniffer\src\Standards\Generic\Sniffs\Functions\OpeningFunctionBraceBsdAllmanSniff.php
                => property "checkFunctions" set to "true"
                => property "checkClosures" set to "false"
        Processing rule "Generic.Functions.OpeningFunctionBraceKernighanRitchie"
                => G:\...\vendor\squizlabs\php_codesniffer\src\Standards\Generic\Sniffs\Functions\OpeningFunctionBraceKernighanRitchieSniff.php
                => property "checkFunctions" set to "false"
                => property "checkClosures" set to "true"
        Processing rule "Generic.Metrics.NestingLevel"
                => G:\...\vendor\squizlabs\php_codesniffer\src\Standards\Generic\Sniffs\Metrics\NestingLevelSniff.php
        Processing rule "Generic.NamingConventions.ConstructorName"
                => G:\...\vendor\squizlabs\php_codesniffer\src\Standards\Generic\Sniffs\NamingConventions\ConstructorNameSniff.php
        Processing rule "Generic.NamingConventions.UpperCaseConstantName"
                => G:\...\vendor\squizlabs\php_codesniffer\src\Standards\Generic\Sniffs\NamingConventions\UpperCaseConstantNameSniff.php
        Processing rule "Generic.PHP.DisallowShortOpenTag"
                => G:\...\vendor\squizlabs\php_codesniffer\src\Standards\Generic\Sniffs\PHP\DisallowShortOpenTagSniff.php
        Processing rule "Generic.PHP.ForbiddenFunctions"
                => G:\...\vendor\squizlabs\php_codesniffer\src\Standards\Generic\Sniffs\PHP\ForbiddenFunctionsSniff.php
        Processing rule "Generic.PHP.LowerCaseConstant"
                => G:\...\vendor\squizlabs\php_codesniffer\src\Standards\Generic\Sniffs\PHP\LowerCaseConstantSniff.php
        Processing rule "Generic.PHP.NoSilencedErrors"
                => G:\...\vendor\squizlabs\php_codesniffer\src\Standards\Generic\Sniffs\PHP\NoSilencedErrorsSniff.php
        Processing rule "Generic.Strings.UnnecessaryStringConcat"
                => G:\...\vendor\squizlabs\php_codesniffer\src\Standards\Generic\Sniffs\Strings\UnnecessaryStringConcatSniff.php
                => property "error" set to "false"
        Processing rule "Generic.WhiteSpace.DisallowTabIndent"
                => G:\...\vendor\squizlabs\php_codesniffer\src\Standards\Generic\Sniffs\WhiteSpace\DisallowTabIndentSniff.php
        Processing rule "Generic.WhiteSpace.ScopeIndent"
                => G:\...\vendor\squizlabs\php_codesniffer\src\Standards\Generic\Sniffs\WhiteSpace\ScopeIndentSniff.php
        Processing rule "MySource.PHP.EvalObjectFactory"
                => G:\...\vendor\squizlabs\php_codesniffer\src\Standards\MySource\Sniffs\PHP\EvalObjectFactorySniff.php
        Processing rule "MySource.PHP.GetRequestData"
                => G:\...\vendor\squizlabs\php_codesniffer\src\Standards\MySource\Sniffs\PHP\GetRequestDataSniff.php
        Processing rule "PEAR.ControlStructures.ControlSignature"
                => G:\...\vendor\squizlabs\php_codesniffer\src\Standards\PEAR\Sniffs\ControlStructures\ControlSignatureSniff.php
        Processing rule "PEAR.ControlStructures.MultiLineCondition"
                => G:\...\vendor\squizlabs\php_codesniffer\src\Standards\PEAR\Sniffs\ControlStructures\MultiLineConditionSniff.php
        Processing rule "PEAR.Files.IncludingFile"
                => G:\...\vendor\squizlabs\php_codesniffer\src\Standards\PEAR\Sniffs\Files\IncludingFileSniff.php
        Processing rule "PEAR.Formatting.MultiLineAssignment"
                => G:\...\vendor\squizlabs\php_codesniffer\src\Standards\PEAR\Sniffs\Formatting\MultiLineAssignmentSniff.php
        Processing rule "PEAR.Functions.FunctionCallSignature"
                => G:\...\vendor\squizlabs\php_codesniffer\src\Standards\PEAR\Sniffs\Functions\FunctionCallSignatureSniff.php
                => property "allowMultipleArguments" set to "false"
        Processing rule "PEAR.Functions.ValidDefaultValue"
                => G:\...\vendor\squizlabs\php_codesniffer\src\Standards\PEAR\Sniffs\Functions\ValidDefaultValueSniff.php
        Processing rule "PEAR.NamingConventions.ValidClassName"
                => G:\...\vendor\squizlabs\php_codesniffer\src\Standards\PEAR\Sniffs\NamingConventions\ValidClassNameSniff.php
        Processing rule "PEAR.WhiteSpace.ObjectOperatorIndent"
                => G:\...\vendor\squizlabs\php_codesniffer\src\Standards\PEAR\Sniffs\WhiteSpace\ObjectOperatorIndentSniff.php
        Processing rule "PEAR.WhiteSpace.ScopeClosingBrace"
                => G:\...\vendor\squizlabs\php_codesniffer\src\Standards\PEAR\Sniffs\WhiteSpace\ScopeClosingBraceSniff.php
        Processing rule "PSR1.Classes.ClassDeclaration"
                => G:\...\vendor\squizlabs\php_codesniffer\src\Standards\PSR1\Sniffs\Classes\ClassDeclarationSniff.php
        Processing rule "PSR2.Classes.ClassDeclaration"
                => G:\...\vendor\squizlabs\php_codesniffer\src\Standards\PSR2\Sniffs\Classes\ClassDeclarationSniff.php
        Processing rule "PSR2.Classes.PropertyDeclaration"
                => G:\...\vendor\squizlabs\php_codesniffer\src\Standards\PSR2\Sniffs\Classes\PropertyDeclarationSniff.php
        Processing rule "PSR2.ControlStructures.ControlStructureSpacing"
                => G:\...\vendor\squizlabs\php_codesniffer\src\Standards\PSR2\Sniffs\ControlStructures\ControlStructureSpacingSniff.php
        Processing rule "PSR2.ControlStructures.ElseIfDeclaration"
                => G:\...\vendor\squizlabs\php_codesniffer\src\Standards\PSR2\Sniffs\ControlStructures\ElseIfDeclarationSniff.php
        Processing rule "PSR2.ControlStructures.SwitchDeclaration"
                => G:\...\vendor\squizlabs\php_codesniffer\src\Standards\PSR2\Sniffs\ControlStructures\SwitchDeclarationSniff.php
        Processing rule "PSR2.Files.EndFileNewline"
                => G:\...\vendor\squizlabs\php_codesniffer\src\Standards\PSR2\Sniffs\Files\EndFileNewlineSniff.php
        Processing rule "PSR2.Methods.MethodDeclaration"
                => G:\...\vendor\squizlabs\php_codesniffer\src\Standards\PSR2\Sniffs\Methods\MethodDeclarationSniff.php
        Processing rule "PSR2.Namespaces.NamespaceDeclaration"
                => G:\...\vendor\squizlabs\php_codesniffer\src\Standards\PSR2\Sniffs\Namespaces\NamespaceDeclarationSniff.php
        Processing rule "PSR2.Namespaces.UseDeclaration"
                => G:\...\vendor\squizlabs\php_codesniffer\src\Standards\PSR2\Sniffs\Namespaces\UseDeclarationSniff.php
        Processing rule "Squiz.PHP.DisallowMultipleAssignments"
                => G:\...\vendor\squizlabs\php_codesniffer\src\Standards\Squiz\Sniffs\PHP\DisallowMultipleAssignmentsSniff.php
        Processing rule "Squiz.PHP.DisallowSizeFunctionsInLoops"
                => G:\...\vendor\squizlabs\php_codesniffer\src\Standards\Squiz\Sniffs\PHP\DisallowSizeFunctionsInLoopsSniff.php
        Processing rule "Squiz.PHP.EmbeddedPhp"
                => G:\...\vendor\squizlabs\php_codesniffer\src\Standards\Squiz\Sniffs\PHP\EmbeddedPhpSniff.php
        Processing rule "Squiz.PHP.Eval"
                => G:\...\vendor\squizlabs\php_codesniffer\src\Standards\Squiz\Sniffs\PHP\EvalSniff.php
        Processing rule "Squiz.PHP.GlobalKeyword"
                => G:\...\vendor\squizlabs\php_codesniffer\src\Standards\Squiz\Sniffs\PHP\GlobalKeywordSniff.php
        Processing rule "Squiz.PHP.InnerFunctions"
                => G:\...\vendor\squizlabs\php_codesniffer\src\Standards\Squiz\Sniffs\PHP\InnerFunctionsSniff.php
        Processing rule "Squiz.PHP.LowercasePHPFunctions"
                => G:\...\vendor\squizlabs\php_codesniffer\src\Standards\Squiz\Sniffs\PHP\LowercasePHPFunctionsSniff.php
        Processing rule "Squiz.PHP.NonExecutableCode"
                => G:\...\vendor\squizlabs\php_codesniffer\src\Standards\Squiz\Sniffs\PHP\NonExecutableCodeSniff.php
        Processing rule "Squiz.Scope.StaticThisUsage"
                => G:\...\vendor\squizlabs\php_codesniffer\src\Standards\Squiz\Sniffs\Scope\StaticThisUsageSniff.php
        Processing rule "Squiz.WhiteSpace.CastSpacing"
                => G:\...\vendor\squizlabs\php_codesniffer\src\Standards\Squiz\Sniffs\WhiteSpace\CastSpacingSniff.php
        Processing rule "Squiz.WhiteSpace.LanguageConstructSpacing"
                => G:\...\vendor\squizlabs\php_codesniffer\src\Standards\Squiz\Sniffs\WhiteSpace\LanguageConstructSpacingSniff.php
        Processing rule "Squiz.WhiteSpace.LogicalOperatorSpacing"
                => G:\...\vendor\squizlabs\php_codesniffer\src\Standards\Squiz\Sniffs\WhiteSpace\LogicalOperatorSpacingSniff.php
        Processing rule "Squiz.WhiteSpace.ObjectOperatorSpacing"
                => G:\...\vendor\squizlabs\php_codesniffer\src\Standards\Squiz\Sniffs\WhiteSpace\ObjectOperatorSpacingSniff.php
        Processing rule "Squiz.WhiteSpace.OperatorSpacing"
                => G:\...\vendor\squizlabs\php_codesniffer\src\Standards\Squiz\Sniffs\WhiteSpace\OperatorSpacingSniff.php
        Processing rule "Squiz.WhiteSpace.SemicolonSpacing"
                => G:\...\vendor\squizlabs\php_codesniffer\src\Standards\Squiz\Sniffs\WhiteSpace\SemicolonSpacingSniff.php
=> Ruleset processing complete; included 60 sniffs and excluded 0
        *** START PHP TOKENIZING ***
        Process token [0]: T_OPEN_TAG => <?php
        Process token [1]: T_DO => do
        Process token [2]: T_WHITESPACE =>
        Process token  3 : T_OPEN_CURLY_BRACKET => {
        Process token [4]: T_CLOSE_TAG => ?>
        *** END PHP TOKENIZING ***
        *** START PHP TOKENIZING ***
        Process token [0]: T_OPEN_TAG => <?php
        Process token [1]: T_WHITESPACE =>
        Process token [2]: T_WHILE => while
        Process token [3]: T_WHITESPACE =>
        Process token  4 : T_OPEN_PARENTHESIS => (
        Process token [5]: T_CLOSE_TAG => ?>
        *** END PHP TOKENIZING ***
        *** START PHP TOKENIZING ***
        Process token [0]: T_OPEN_TAG => <?php
        Process token  1 : T_SEMICOLON => ;
        Process token [2]: T_CLOSE_TAG => ?>
        *** END PHP TOKENIZING ***
        *** START PHP TOKENIZING ***
        Process token [0]: T_OPEN_TAG => <?php
        Process token [1]: T_WHILE => while
        Process token [2]: T_WHITESPACE =>
        Process token  3 : T_OPEN_PARENTHESIS => (
        Process token [4]: T_CLOSE_TAG => ?>
        *** END PHP TOKENIZING ***
        *** START PHP TOKENIZING ***
        Process token [0]: T_OPEN_TAG => <?php
        Process token [1]: T_WHITESPACE =>
        Process token  2 : T_OPEN_CURLY_BRACKET => {
        Process token [3]: T_CLOSE_TAG => ?>
        *** END PHP TOKENIZING ***
        *** START PHP TOKENIZING ***
        Process token [0]: T_OPEN_TAG => <?php
        Process token [1]: T_FOR => for
        Process token [2]: T_WHITESPACE =>
        Process token  3 : T_OPEN_PARENTHESIS => (
        Process token [4]: T_CLOSE_TAG => ?>
        *** END PHP TOKENIZING ***
        *** START PHP TOKENIZING ***
        Process token [0]: T_OPEN_TAG => <?php
        Process token [1]: T_WHITESPACE =>
        Process token  2 : T_OPEN_CURLY_BRACKET => {
        Process token [3]: T_CLOSE_TAG => ?>
        *** END PHP TOKENIZING ***
        *** START PHP TOKENIZING ***
        Process token [0]: T_OPEN_TAG => <?php
        Process token [1]: T_IF => if
        Process token [2]: T_WHITESPACE =>
        Process token  3 : T_OPEN_PARENTHESIS => (
        Process token [4]: T_CLOSE_TAG => ?>
        *** END PHP TOKENIZING ***
        *** START PHP TOKENIZING ***
        Process token [0]: T_OPEN_TAG => <?php
        Process token [1]: T_WHITESPACE =>
        Process token  2 : T_OPEN_CURLY_BRACKET => {
        Process token [3]: T_CLOSE_TAG => ?>
        *** END PHP TOKENIZING ***
        *** START PHP TOKENIZING ***
        Process token [0]: T_OPEN_TAG => <?php
        Process token [1]: T_FOREACH => foreach
        Process token [2]: T_WHITESPACE =>
        Process token  3 : T_OPEN_PARENTHESIS => (
        Process token [4]: T_CLOSE_TAG => ?>
        *** END PHP TOKENIZING ***
        *** START PHP TOKENIZING ***
        Process token [0]: T_OPEN_TAG => <?php
        Process token [1]: T_WHITESPACE =>
        Process token  2 : T_OPEN_CURLY_BRACKET => {
        Process token [3]: T_CLOSE_TAG => ?>
        *** END PHP TOKENIZING ***
        *** START PHP TOKENIZING ***
        Process token [0]: T_OPEN_TAG => <?php
        Process token  1 : T_CLOSE_CURLY_BRACKET => }
        Process token [2]: T_WHITESPACE =>
        Process token [3]: T_ELSE => else
        Process token [4]: T_WHITESPACE =>
        Process token [5]: T_IF => if
        Process token [6]: T_WHITESPACE =>
        Process token  7 : T_OPEN_PARENTHESIS => (
        Process token [8]: T_CLOSE_TAG => ?>
        *** END PHP TOKENIZING ***
        *** START PHP TOKENIZING ***
        Process token [0]: T_OPEN_TAG => <?php
        Process token [1]: T_WHITESPACE =>
        Process token  2 : T_OPEN_CURLY_BRACKET => {
        Process token [3]: T_CLOSE_TAG => ?>
        *** END PHP TOKENIZING ***
        *** START PHP TOKENIZING ***
        Process token [0]: T_OPEN_TAG => <?php
        Process token  1 : T_CLOSE_CURLY_BRACKET => }
        Process token [2]: T_WHITESPACE =>
        Process token [3]: T_ELSEIF => elseif
        Process token [4]: T_WHITESPACE =>
        Process token  5 : T_OPEN_PARENTHESIS => (
        Process token [6]: T_CLOSE_TAG => ?>
        *** END PHP TOKENIZING ***
        *** START PHP TOKENIZING ***
        Process token [0]: T_OPEN_TAG => <?php
        Process token [1]: T_WHITESPACE =>
        Process token  2 : T_OPEN_CURLY_BRACKET => {
        Process token [3]: T_CLOSE_TAG => ?>
        *** END PHP TOKENIZING ***
        *** START PHP TOKENIZING ***
        Process token [0]: T_OPEN_TAG => <?php
        Process token  1 : T_CLOSE_CURLY_BRACKET => }
        Process token [2]: T_WHITESPACE =>
        Process token [3]: T_ELSE => else
        Process token [4]: T_WHITESPACE =>
        Process token  5 : T_OPEN_CURLY_BRACKET => {
        Process token [6]: T_CLOSE_TAG => ?>
        *** END PHP TOKENIZING ***
        *** START PHP TOKENIZING ***
        Process token [0]: T_OPEN_TAG => <?php
        Process token [1]: T_DO => do
        Process token [2]: T_WHITESPACE =>
        Process token  3 : T_OPEN_CURLY_BRACKET => {
        Process token [4]: T_CLOSE_TAG => ?>
        *** END PHP TOKENIZING ***
Creating file list... DONE (1 files in queue)
Changing into directory G:\...\app\Helpers
Processing test.php
        *** START PHP TOKENIZING ***
        Process token [0]: T_OPEN_TAG => <?php\r\n
        Process token [1]: T_WHITESPACE => \r\n
        Process token [2]: T_STRING => array_reduce
        Process token  3 : T_OPEN_PARENTHESIS => (
        Process token [4]: T_VARIABLE => $array
        Process token  5 : T_COMMA => ,
        Process token [6]: T_WHITESPACE =>
        Process token [7]: T_FUNCTION => function
        Process token [8]: T_WHITESPACE =>
        Process token  9 : T_OPEN_PARENTHESIS => (
        Process token [10]: T_VARIABLE => $carry
        Process token  11 : T_COMMA => ,
        Process token [12]: T_WHITESPACE =>
        Process token [13]: T_VARIABLE => $item
        Process token  14 : T_CLOSE_PARENTHESIS => )
        Process token [15]: T_WHITESPACE =>
        Process token [16]: T_USE => use
        Process token [17]: T_WHITESPACE =>
        Process token  18 : T_OPEN_PARENTHESIS => (
        Process token [19]: T_VARIABLE => $variable
        Process token  20 : T_CLOSE_PARENTHESIS => )
        Process token [21]: T_WHITESPACE =>
        Process token  22 : T_OPEN_CURLY_BRACKET => {
        Process token [23]: T_WHITESPACE => \r\n
        Process token [24]: T_RETURN => return
        Process token [25]: T_WHITESPACE =>
        Process token [26]: T_STRING => true
        Process token  27 : T_SEMICOLON => ;
        Process token [28]: T_WHITESPACE => \r\n
        Process token  29 : T_CLOSE_CURLY_BRACKET => }
        Process token  30 : T_CLOSE_PARENTHESIS => )
        Process token  31 : T_SEMICOLON => ;
        Process token [32]: T_WHITESPACE => \r\n
        *** END PHP TOKENIZING ***
        *** START TOKEN MAP ***
        => Found curly bracket opener at 22
        => Found curly bracket closer at 30 for 22
        *** END TOKEN MAP ***
        *** START SCOPE MAP ***
        Start scope map at 7:T_FUNCTION => function
        => Begin scope map recursion at token 7 with depth 1
        Process token 8 on line 3 []: T_WHITESPACE =>
        Process token 9 on line 3 []: T_OPEN_PARENTHESIS => (
        Process token 10 on line 3 []: T_VARIABLE => $carry
        Process token 11 on line 3 []: T_COMMA => ,
        Process token 12 on line 3 []: T_WHITESPACE =>
        Process token 13 on line 3 []: T_VARIABLE => $item
        Process token 14 on line 3 []: T_CLOSE_PARENTHESIS => )
        Process token 15 on line 3 []: T_WHITESPACE =>
        Process token 16 on line 3 []: T_USE => use
        Process token 17 on line 3 []: T_WHITESPACE =>
        Process token 18 on line 3 []: T_OPEN_PARENTHESIS => (
        Process token 19 on line 3 []: T_VARIABLE => $variable
        Process token 20 on line 3 []: T_CLOSE_PARENTHESIS => )
        Process token 21 on line 3 []: T_WHITESPACE =>
        Process token 22 on line 3 []: T_OPEN_CURLY_BRACKET => {
        => Found scope opener for 7:T_FUNCTION
        Process token 23 on line 3 [opener:22;]: T_WHITESPACE => \r\n
        Process token 24 on line 4 [opener:22;]: T_WHITESPACE =>
        Process token 25 on line 4 [opener:22;]: T_RETURN => return
        Process token 26 on line 4 [opener:22;]: T_WHITESPACE =>
        Process token 27 on line 4 [opener:22;]: T_TRUE => true
        Process token 28 on line 4 [opener:22;]: T_SEMICOLON => ;
        Process token 29 on line 4 [opener:22;]: T_WHITESPACE => \r\n
        Process token 30 on line 5 [opener:22;]: T_CLOSE_CURLY_BRACKET => }
        => Found scope closer (30:T_CLOSE_CURLY_BRACKET) for 7:T_FUNCTION
        *** END SCOPE MAP ***
        *** START LEVEL MAP ***
        Process token 0 on line 1 [col:1;len:5;lvl:0;]: T_OPEN_TAG => <?php\r\n
        Process token 1 on line 2 [col:1;len:0;lvl:0;]: T_WHITESPACE => \r\n
        Process token 2 on line 3 [col:1;len:12;lvl:0;]: T_STRING => array_reduce
        Process token 3 on line 3 [col:13;len:1;lvl:0;]: T_OPEN_PARENTHESIS => (
        Process token 4 on line 3 [col:14;len:6;lvl:0;]: T_VARIABLE => $array
        Process token 5 on line 3 [col:20;len:1;lvl:0;]: T_COMMA => ,
        Process token 6 on line 3 [col:21;len:1;lvl:0;]: T_WHITESPACE =>
        Process token 7 on line 3 [col:22;len:8;lvl:0;]: T_FUNCTION => function
        Process token 8 on line 3 [col:30;len:1;lvl:0;]: T_WHITESPACE =>
        Process token 9 on line 3 [col:31;len:1;lvl:0;]: T_OPEN_PARENTHESIS => (
        Process token 10 on line 3 [col:32;len:6;lvl:0;]: T_VARIABLE => $carry
        Process token 11 on line 3 [col:38;len:1;lvl:0;]: T_COMMA => ,
        Process token 12 on line 3 [col:39;len:1;lvl:0;]: T_WHITESPACE =>
        Process token 13 on line 3 [col:40;len:5;lvl:0;]: T_VARIABLE => $item
        Process token 14 on line 3 [col:45;len:1;lvl:0;]: T_CLOSE_PARENTHESIS => )
        Process token 15 on line 3 [col:46;len:1;lvl:0;]: T_WHITESPACE =>
        Process token 16 on line 3 [col:47;len:3;lvl:0;]: T_USE => use
        Process token 17 on line 3 [col:50;len:1;lvl:0;]: T_WHITESPACE =>
        Process token 18 on line 3 [col:51;len:1;lvl:0;]: T_OPEN_PARENTHESIS => (
        Process token 19 on line 3 [col:52;len:9;lvl:0;]: T_VARIABLE => $variable
        Process token 20 on line 3 [col:61;len:1;lvl:0;]: T_CLOSE_PARENTHESIS => )
        Process token 21 on line 3 [col:62;len:1;lvl:0;]: T_WHITESPACE =>
        Process token 22 on line 3 [col:63;len:1;lvl:0;]: T_OPEN_CURLY_BRACKET => {
        => Found scope opener for 7:T_FUNCTION
                * level increased *
                * token 7:T_FUNCTION added to conditions array *
                Process token 23 on line 3 [col:64;len:0;lvl:1;conds;T_FUNCTION;]: T_WHITESPACE => \r\n
                Process token 24 on line 4 [col:1;len:4;lvl:1;conds;T_FUNCTION;]: T_WHITESPACE =>
                Process token 25 on line 4 [col:5;len:6;lvl:1;conds;T_FUNCTION;]: T_RETURN => return
                Process token 26 on line 4 [col:11;len:1;lvl:1;conds;T_FUNCTION;]: T_WHITESPACE =>
                Process token 27 on line 4 [col:12;len:4;lvl:1;conds;T_FUNCTION;]: T_TRUE => true
                Process token 28 on line 4 [col:16;len:1;lvl:1;conds;T_FUNCTION;]: T_SEMICOLON => ;
                Process token 29 on line 4 [col:17;len:0;lvl:1;conds;T_FUNCTION;]: T_WHITESPACE => \r\n
                Process token 30 on line 5 [col:1;len:1;lvl:1;conds;T_FUNCTION;]: T_CLOSE_CURLY_BRACKET => }
                => Found scope closer for 22:T_OPEN_CURLY_BRACKET
                * token T_FUNCTION removed from conditions array *
                * level decreased *
        Process token 31 on line 5 [col:2;len:1;lvl:0;]: T_CLOSE_PARENTHESIS => )
        Process token 32 on line 5 [col:3;len:1;lvl:0;]: T_SEMICOLON => ;
        Process token 33 on line 5 [col:4;len:0;lvl:0;]: T_WHITESPACE => \r\n
        *** END LEVEL MAP ***
        *** START ADDITIONAL PHP PROCESSING ***
        * token 7 on line 3 changed from T_FUNCTION to T_CLOSURE
                * cleaned 23 (T_WHITESPACE) *
                * cleaned 24 (T_WHITESPACE) *
                * cleaned 25 (T_RETURN) *
                * cleaned 26 (T_WHITESPACE) *
                * cleaned 27 (T_TRUE) *
                * cleaned 28 (T_SEMICOLON) *
                * cleaned 29 (T_WHITESPACE) *
        *** END ADDITIONAL PHP PROCESSING ***
[PHP => 34 tokens in 5 lines]...
DONE in 192ms (3 errors, 0 warnings)

FILE: G:\...\app\Helpers\test.php
----------------------------------------------------------------------
FOUND 3 ERRORS AFFECTING 2 LINES
----------------------------------------------------------------------
 3 | ERROR | [x] Opening parenthesis of a multi-line function call
   |       |     must be the last content on the line
   |       |     (PEAR.Functions.FunctionCallSignature.ContentAfterOpenBracket)
 3 | ERROR | [x] Only one argument is allowed per line in a
   |       |     multi-line function call
   |       |     (PEAR.Functions.FunctionCallSignature.MultipleArguments)
 5 | ERROR | [x] Closing parenthesis of a multi-line function call
   |       |     must be on a line by itself
   |       |     (PEAR.Functions.FunctionCallSignature.CloseBracketLine)
----------------------------------------------------------------------
PHPCBF CAN FIX THE 3 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------

Time: 559ms; Memory: 8Mb
```
